src/Symfony/Component/Config/Definition/Builder/NodeFinder.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config\Definition\Builder;
+
+/**
+ * @author Jan Schädlich <jan.schaedlich@sensiolabs.de>
+ */
+class NodeFinder
+{
+    public function find(string $nodePath, NodeDefinition $rootNode)
+    {
+        if (!$rootNode instanceof ArrayNodeDefinition) {
+            if ($nodePath !== $rootNode->getName()) {
+                throw new \RuntimeException(
+                    sprintf(
+                        'Node with name "%s" does not exist in the given root node "%s"!',
+                        $nodePath,
+                        $rootNode->getName()
+                    )
+                );
+            }
+
+            return $rootNode;
+        }
+
+        $explodedNodePaths = explode('.', $nodePath);
+
+        if (!array_key_exists($explodedNodePaths[0], $rootNode->getChildNodeDefinitions())) {
+            throw new \RuntimeException(
```

Comment:
The whole exception should be on one line, Symfony uses long lines CS

Reply:
Fixed